### ISSUE TITLE
Node restarts generates some errors that seem transient and not affecting anything (spams slack as a result)

### ISSUE THREAD

**Issue by ilya-git** [fluxcd/flux2#1969]
### Describe the bug

I have installed `kured` into the cluster which restarts nodes from time to time. When it does, very often HelmChart generates these events in k8s:

```
Events:
  Type    Reason  Age   From               Message
  Normal  error   90m   source-controller  open /data/helmrepository/flux-system/kured/index-553ef5d828e6edb6334598154c88f31ff6b3f0b7.yaml: no such file or directory
  Normal  info    90m   source-controller  Fetched revision: 2.10.0
```

This happens not only to `kured` itself, but it seems rather to random helm charts. This is also reflected in our slack via notification controller.

This error seem like either a bug or rather some internal details that should not propagate further (since it does not seem anything fails, there is a an event that the revision was fetched right after).

Is it possible to do something so that these errors won't be visible in both k8s and notification?

P.S. There are multiple errors of different kind that seem to happen when a node restarts:

1. `imagepolicy/my-image.flux-system` version list argument cannot be empty
2. `helmchart/my-chart.flux-system` open /data/helmrepository/flux-system/my-chart/index-47ec672098c8bbdacd5209b86f21f6bb8870175e.yaml: no such file or directory
3. `helmchart/my-chart.flux-system` artifact open error: open /data/gitrepository/flux-system/my-gitops/5bc5bd2eecc46fe3188a6478cac20e4be5358e23.tar.gz: no such file or directory

### Steps to reproduce

I am actually not quite sure how to reproduce it, but I suppose draining nodes that run some flux components will do the trick while having some helm charts installed.

I can try to find out more details if necessary.

### Expected behavior

No "transient" errors as notification, no error in k8s.

### OS / Distro

Default AKS nodes (Ubuntu 18.04.6 LTS)

### Flux version

v0.18.3

### Flux check

► checking prerequisites
✗ flux 0.18.3 <0.19.0 (new version is available, please upgrade)
✔ Kubernetes 1.21.2 >=1.19.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.12.0
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.15.0
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.12.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.15.5
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.17.1
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.16.0
✔ all checks passed

### Git provider

Bitbucket cloud

### Container Registry provider

Azure Container Registry

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aliusmiles** [comment]
I've tried to scale source-controller to 2 replicas, but apparently second replica is not even starting controllers until it's became leader so when I've killed current source-controller leader pod - still received bunch of notifications. I guess having stand-by replica pulling all the sources could lead to conflicts?

I've migrated my apps from helm to kustomize and now have much less "noise" messages from Flux :) Now receiving those only for "system" services, still deployed by helm, like prometheus/etc. Thinking about migrating those to kustomize as well, as it looks like it works so much better with Flux then helm..

**Comment by jtnz** [comment]
We're also seeing the `imagepolicy/foo.flux version list argument cannot be empty` error spam. We run mostly on spot instances in AWS so I'm guessing the pod is just being moved. To me it sounds like there could be a (simple?) code change to not complain about the `version list argument cannot be empty` when it hasn't first scanned the repo? We have hardly any volumes and it feels like a hack to have to put one in place for this (EBS volumes also require `recreate` for the rollout strategy).

**Comment by stefanprodan** [comment]
404 errors and `version list argument cannot be empty` should be fixed in Flux 0.32.0
